doc/man3/DH_new_by_nid.pod

```diff
@@ -21,7 +21,11 @@ B<NID_modp_4096>, B<NID_modp_6144> or B<NID_modp_8192>.
 
 DH_get_nid() determines if the parameters contained in B<dh> match
 any named set. It returns the NID corresponding to the matching parameters or
-B<NID_undef> if there is no match.
+B<NID_undef> if there is no match. Internally it caches the nid, so that any
+subsequent calls can fetch the cached value.
+Note that if the value of parameter q is not set, then it is set to q = p * 2 + 1.
```

Comment:
I assume that q is set to (p -1)/ 2. But I assume it would only do that in case it's a safe prime, which just happens to be the case.

Reply:
ok I see your point I will change it to be correct and put a comment about them all being safe primes.